spyder/plugins/help/utils/conf.py

```diff
@@ -59,7 +59,7 @@
 jsmath_path = 'easy/load.js'
 
 # The suffix of source filenames.
-source_suffix = '.rst'
+source_suffix = {".rst": "restructuredtext"}
```

Comment:
According to the Sphinx docs you referenced, this was introduced in Sphinx `1.8.0`, so you also need to update our minimal Sphinx requirement (which at moment is `0.6.6`). However, given that we now support Python 3.9+, that needs to be `>=7.2.0` according to the [Sphinx changelog](https://www.sphinx-doc.org/en/master/changes/7.2.html#release-7-2-0-released-aug-17-2023).

You need to do that in `setup.py`, `requirements/main.yml`, `binder/environment.yml` and `spyder/dependencies.py`.

Reply:
Done.